### ISSUE TITLE
Feature: Static Asserts

### ISSUE THREAD

**Issue by scott-wilson** [vfx-rs/cppmm#1]
Right now the C++ side of the bindings doesn't generate static asserts. At minimum, we should have code as follows:

```cpp
CPPMM_DEFINE_POINTER_CASTS(Imath_3_0::half, Imath_3_0_half)

static_assert(sizeof(Imath_3_0::half) == sizeof(Imath_3_0_half));
static_assert(alignof(Imath_3_0::half) == alignof(Imath_3_0_half));

extern "C" {
...
}
```

If we support having POD structs on either side of the binding, then we'll want to also validate the struct's fields offsets as well.

**Comment by anderslanglands** [comment]
Already done  in my local copy (including offset of fields). Will push
later today.

**Comment by scott-wilson** [comment]
Cool, I'll wait on more notes until the next push.
